cmd/collector/app/http_handler_test.go

```diff
@@ -35,20 +36,30 @@ import (
 	tchanThrift "github.com/uber/tchannel-go/thrift"
 
 	"github.com/uber/jaeger/thrift-gen/jaeger"
-	tJaeger "github.com/uber/jaeger/thrift-gen/jaeger"
 	"github.com/uber/jaeger/thrift-gen/zipkincore"
 )
 
 var httpClient = &http.Client{Timeout: 2 * time.Second}
 
 type mockJaegerHandler struct {
-	err error
+	err     error
+	mux     sync.Mutex
+	batches []*jaeger.Batch
 }
 
 func (p *mockJaegerHandler) SubmitBatches(ctx tchanThrift.Context, batches []*jaeger.Batch) ([]*jaeger.BatchSubmitResponse, error) {
+	p.mux.Lock()
+	defer p.mux.Unlock()
+	p.batches = append(p.batches, batches...)
 	return nil, p.err
 }
 
+func (p *mockJaegerHandler) getBatches() []*jaeger.Batch {
+	p.mux.Lock()
+	defer p.mux.Unlock()
```

Comment:
Do you need this lock for reading? You already have one on writing, so, there's no risk on getting an inconsistent `batches` here, right?